### ISSUE TITLE
ToDo: AS overhaul

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#528]
To do after we get 63-beta out

- move it out of system addons section (no need to cater for ESR, we have an archived v60 for that)
- expand/check on all the prefs (although old, see #396 )

**Comment by claustromaniac** [comment]
FWIW I have this on FF 64. Some of them are absolutely not necessary, but whatever.
<details><summary> clickey </summary><p>



```js
// ACTIVITY STREAM
user_pref("browser.newtabpage.activity-stream.asrouter.devtoolsEnabled", false);
user_pref("browser.newtabpage.activity-stream.default.sites", "");
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.aboutpreferences", false); // FF61+
user_pref("browser.newtabpage.activity-stream.feeds.asrouterfeed", false);
user_pref("browser.newtabpage.activity-stream.feeds.favicon", false);
user_pref("browser.newtabpage.activity-stream.feeds.messagecenterfeed", false); // FF61+
user_pref("browser.newtabpage.activity-stream.feeds.migration", false);
user_pref("browser.newtabpage.activity-stream.feeds.newtabinit", false);
user_pref("browser.newtabpage.activity-stream.feeds.places", false);
user_pref("browser.newtabpage.activity-stream.feeds.prefs", false);
user_pref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories.options", "");
user_pref("browser.newtabpage.activity-stream.feeds.sections", false);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false);
user_pref("browser.newtabpage.activity-stream.feeds.systemtick", false);
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.feeds.theme", false);
user_pref("browser.newtabpage.activity-stream.feeds.topsites", false);
user_pref("browser.newtabpage.activity-stream.filterAdult", false);
user_pref("browser.newtabpage.activity-stream.fxaccounts.endpoint", "");
user_pref("browser.newtabpage.activity-stream.improvesearch.noDefaultSearchTile", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.searchEngines", "");
user_pref("browser.newtabpage.activity-stream.messageCenterExperimentEnabled", false);
// user_pref("browser.newtabpage.activity-stream.migrationExpired", true);
user_pref("browser.newtabpage.activity-stream.prerender", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeBookmarks", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeDownloads", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeVisited", false);
user_pref("browser.newtabpage.activity-stream.showSearch", false);
user_pref("browser.newtabpage.activity-stream.showSponsored", false);
user_pref("browser.newtabpage.activity-stream.showTopSites", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
user_pref("browser.newtabpage.activity-stream.telemetry.ut.events", false);
user_pref("browser.newtabpage.activity-stream.tippyTop.service.endpoint", "");
```


</p></details>

**Comment by Thorin-Oakenpants** [comment]
^^ thanks. I updated to 63 about 12 hours ago, and only using the current js ones, I noticed that highlights (master option) and the four suboptions in about:preferences#home weren't unchecked. Not that it matters right now, but I think at least listing all the options would be good (these can definitely fall under shoulder surfers, I don't want my goat & llama porn sites prominently displayed, tyvm).

After beta (yeah I know, it'll happen), I will play with a nilla and see what else may be required to stop any persistent caching. Top sites fall under history AFAIK, and pocket is already catered for.

**Comment by earthlng** [comment]
> listing all the options would be good

we already use a blank newtab page and disable the remote-y things because that's what most people don't like. If someone wants to use AS they can easily find all remaining options in the UI.
I don't think we need to add anything more atm.

**Comment by Thorin-Oakenpants** [comment]
where would we put this? I think `0110` would be nice. Something like

```js
/* 0110: clean up Activity Stream (AS)
 * AS is the default homepage/newtab in FF57+, based on metadata and browsing behavior.
 * [SETTING] Home>Firefox Home Content>...  ***/
/* 0111: disable AS Web Search ***/
   // user_pref("browser.newtabpage.activity-stream.showSearch", false);
/* 0112: disable AS Top Sites ***/
user_pref("browser.newtabpage.activity-stream.feeds.topsites", false)
/* 0113: disable AS Pocket and Sponsored Stories ***/
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false)
user_pref("browser.newtabpage.activity-stream.showSponsored", false)
/* 0114: disable Highlights ***/
user_pref("browser.newtabpage.activity-stream.feeds.section.highlights", false)
user_pref("browser.newtabpage.activity-stream.section.highlights.includeVisited", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeBookmarks", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeDownloads", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false);
/* 0115 disable AS Snippets ***/
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false);

   // numbers gap for future shit

/* 0120: disable AS telemetry ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");

/* 0121: disable AS library entries? or whatever this was ***/
user_pref("browser.library.activity-stream.enabled", false); // (FF57+)
```

There are still quite a few other items. Disabling/toggling the display of sections is one thing, but continuing to store data is another - the `browser.newtabpage.activity-stream.tippyTop.service.endpoint` was one that was important if you wanted FF to "be quiet".

Anyway, there's all the UI options mapped in the same order in sections 0111 to 1115. It's a start. Nits?

**Comment by Thorin-Oakenpants** [comment]
If onboarding and pocket (and others in future?) are going to be removed from system addons and integrated directly in Firefox, then maybe we should rethink where this goes?

**Comment by Thorin-Oakenpants** [comment]
> If someone wants to use AS they can easily find all remaining options in the UI. I don't think we need to add anything more atm.

I'm in two minds. Users can just go to Options and uncheck everything. So lets remove ALL of those to slim this down. And all we need is to add any other entries: telemetry / connections / urls / continued data storage. And then do we make them active/inactive (because we are no longer disabling AS for end users)?

As it currently stands, we have 9 prefs. The revamp above pushes it to 14 and completes everything in the UI. AFAIC, it's list em all (UI), or list none. I'm fine with either.

```js
/* 0110: clean up Activity Stream (AS) and related telemetry
 * AS is the default homepage/newtab in FF57+, based on metadata and browsing behavior.
 * Use the [SETTING] Home>Firefox Home Content>...  to show/hide what you want ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
/* 0111: if you do not use AS then you might want these ***/
   // user_pref("browser.library.activity-stream.enabled", false);
   // more shit
```

In fact I'm really leaning towards the above. Speak now or forever hold ~my penis~ your peace! :grinning:

**Comment by Thorin-Oakenpants** [comment]
Also, is AS telemetry governed by the telemetry master switch - which in the UI is datareporting.healthreport.uploadEnabled (0333) > Privacy & Security>Firefox Data Collection & Use>Allow Firefox to send technical and **interaction* data . IMO it should be. Would be nice to check if that's the case (imagine the shitstorm if it wasn't) and add that info.

**Comment by Thorin-Oakenpants** [comment]
gee .. the silence is overwhelming

**Comment by Atavic** [comment]
Seems another kind of beast to me.
It has its own modules, type:
```resource://activity-stream/```

**Comment by earthlng** [comment]
```js
/* 0105: disable Activity Stream stuff (FF54+)
 * Activity Stream is the default homepage/newtab in FF57+. It is based on metadata and browsing behavior,
 * and includes telemetry and web content such as snippets, top stories (pocket), top sites, etc.
 *  - ONE: make sure to set your "home" and "newtab" to about:blank (or use an extension to control them)
 *  - TWO: DELETE the XPI file in your System Add-ons directory (note this get reinstalled on app updates)
 * And/or you can try to control the ever-growing, ever-changing "browser.newtabpage.activity-stream.*" prefs
 * [FF63+] Activity Stream (AS) is now builtin and no longer an easily deletable system addon!
 * [SETTING] Home>Firefox Home Content> ...
 * [1] https://wiki.mozilla.org/Firefox/Activity_Stream ***/
/* 0105a: disable AS telemetry ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
/* 0105b: disable AS Snippets ***/
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false); // [SETTING] Home>Firefox Home Content>Snippets
/* 0105c: disable AS Top Stories and other Pocket-based and/or sponsored content ***/
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false); // [SETTING] Home>Firefox Home Content>Highlights>Pages Saved to Pocket
user_pref("browser.newtabpage.activity-stream.showSponsored", false);
/* 0105d: disable AS recent Highlights in the Library (FF57+) ***/
   // user_pref("browser.library.activity-stream.enabled", false);
```

**Comment by earthlng** [comment]
or "disable Activity Stream shit (ASS)" and "disable ASS telemetry" etc :smile:

**Comment by Thorin-Oakenpants** [comment]
I don't like it, E. Couple of lines get too long, and there's way too much text, explaining it - it's FF63. AS has been around for 7 releases so I don't think we need to be too verbose (and we can drop the XPI lines etc). Ref [1] looks like a pretty useless page, so I'd drop that. And it's right under 0103 + 0104 which both say (see 0514), so it's already all tied in - no need to explain setting those, IMO.

```js
/* 0110: clean up Activity Stream (AS) and related telemetry (do we really want FF54+ here)
 * AS is the default homepage/newtab in FF57+, based on metadata and browsing behavior.
 * line about [SETTING] Home>Firefox Home Content>...  to show/hide what you want ***/
```

Isn't that so much better? Can we agree on that ^^ ... 3 lines?

---

So that leaves what prefs we put in.  We have three choices AFAICT
1. list em all (we want privacy from shoulder surfers etc, kill the lot)
2. list only items that make background connections (which is what I think you're angling at)
3. list nothing from the UI, because it keeps it super clean and using the UI is child's play, and it's less maintenance, etc (this would mean dropping the snippets and pocket prefs, which is no big deal, because the whole AS is a shoulder surfing nightmare)

I'm open to suggestions. If we go with option 2, then I'd rather just list em as one lump e.g.

```js
/* 0105: clean up Activity Stream (AS) and related telemetry
 * AS is the default homepage/newtab in FF57+, based on metadata and browsing behavior.
 * [SETTING] Home>Firefox Home Content>...  to show/hide what you want
 * Below are **SOME** settings we think most people would want disabled ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
etc
```

Anyone else got any preferences on this? Last time we were in favor of listing nothing and just saying delete the XPI. Personally, I'm in favor of option 3

**Comment by claustromaniac** [comment]
# :cat2:'s :ballot_box: = :hash::three:

However, as long as the remote stuff stays disabled, I don't really mind. If you end up going for option 1, you may want to add the ui stuff inactive, though. Or, maybe that would be better left for the ever in-progress relaxed variant? BTW, and this is outright OT, what do you guys think about asking for more feedback? Adding some lines in the wiki and/or maybe in the updater scripts might help with that.

**Comment by Thorin-Oakenpants** [comment]
Nice emoji work there :cat2: 

Not sure re asking for contribs/feedback. I def wouldn't add anything to the updater scripts (without looking I assume it has a link to this repo). The user.js already lists this repo (and surely people can see that its github and they can open issues and ask questions).

**Comment by earthlng** [comment]
we can go with the 3-line header but I don't really like your title.
I'm in favor of option 2 and NOT lumped together. 

btw those who want AS real pretty may also want to reset `browser.pagethumbnails.capturing_disabled`.

**Comment by Thorin-Oakenpants** [comment]
> I don't really like your title

sure. I'll leave it as was `disable Activity Stream`. Not putting a version, besides the next line says 57+ for when it was turned on.

> btw those who want AS real pretty may also want to reset browser.pagethumbnails.capturing_disabled

good point, maybe I can add that in

---

I don't think anyone wants option 1: It wouldn't be hard to maintain, but its bloat, and it's messy as in what state we have: end-users will all want something different and end up having to add shit to overrides.

The only difference between options 2 and 3, is that option 2 has two extra prefs (which are in the UI):
- Showing pages "saved to pocket stuff", which is already covered by `0510`. If users have pocket & AS enabled, then, why create another roadblock for them? You would already have a pocket account and be syncing data from the cloud already anyway.
- Snippets. IDK, maybe some people like this shit, at least one person, right? Anyway, it's trivial. It's not a security or privacy or tracking risk. Not 100% sure on when it updates local snippet data or if it pulls if off a server each time - and I don't really care if we point them to using the UI.

If you can agree that these two prefs are not that important, then I think we're **all** (:jeans: , :cat2:  and yourself) set to go option 3 which is the least work and maintenance

don't fight me on this bro :) despite the picture, it's currently 2 on 1 (:cat2: is just out of camera in a cardboard tube)
![dontfightmebro](https://user-images.githubusercontent.com/16656956/48680984-7f1d0f00-eb97-11e8-8862-394bec13f87b.png)

**Comment by Thorin-Oakenpants** [comment]
also: AS telemetry could go under 0300s with other crash reports, telemetry stuff. this would keep startup smaller and cleaner

```js
/* 0331: disable Activity Stream telemetry ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
```

**Comment by Thorin-Oakenpants** [comment]
Hmm, pocket I can do without. But snippets being two prefs is annoying (and they should go together). I think I have a workaround.

**Comment by Thorin-Oakenpants** [comment]
OK, so I'm thinking about possibly considering perhaps changing my mind (I'm allowed to), but this still needs MOAR thought (because there's other settings to think about). In order to not piss about with having snippets inactive, we include and make both active (seriously, no-one wants snippets). So now I'm leaning towards Option 2, because some other UI options are also "pushed" (sponsored stories etc), and there are other prefs which will affect these (still working them out, but eg connection & API info to pocket)

Trying to dis-entangle the UI items (which is only show/hide) from others (which are to stop the background connections and storage of local data) is a mess.

I'm ending up with the same as Big E (except I do not want long [SETTING] strings after prefs). But there's more to add. So I'll do a commit to get this out of the way and then we can discuss what else to add

**Comment by Thorin-Oakenpants** [comment]
`browser.newtabpage.activity-stream.feeds.section.topstories.options` could go in - it's the pocket info. I'd rather kill this stuff, regardless of whether or not any "master" pref is actually a master pref, because it's just like the devs to want everything grabbed on timers and stored locally for faster results/display.

```
{"api_key_pref":"extensions.pocket.oAuthConsumerKey","hidden":false,"provider_icon":"pocket","provider_name":"Pocket","read_more_endpoint":"https://getpocket.com/explore/trending?src=fx_new_tab","stories_endpoint":"https://getpocket.cdn.mozilla.net/v3/firefox/global-recs?version=3&consumer_key=$apiKey&locale_lang=en-US&feed_variant=default_spocs_on","stories_referrer":"https://getpocket.com/recommendations","topics_endpoint":"https://getpocket.cdn.mozilla.net/v3/firefox/trending-topics?version=2&consumer_key=$apiKey&locale_lang=en-US","show_spocs":true,"personalized":true}
```

I also have questions about this shit in devtools :
user_pref("browser.newtabpage.activity-stream.asrouter.messageProviders", "")
   // ^^ not in claustro's lot, but this is snippets etc for devtools - including **SNIPPETS** connection info?
user_pref("browser.newtabpage.activity-stream.asrouter.devtoolsEnabled", false);
user_pref("browser.newtabpage.activity-stream.feeds.asrouterfeed", false);

I'm also concerned with favicons
user_pref("browser.newtabpage.activity-stream.feeds.favicon", false);

And I'm also wondering about the default search shortcuts for google, amazon

Haven't looked at too much more. I wish there was a fuckin master switch

**Comment by Thorin-Oakenpants** [comment]
If anyone cares about all the other prefs, and the new ones in 64 diffs which start to build in onboarding into some of them, then feel free to reopen and contribute. I've had enough of having to do every-F-thing myself

**Comment by Thorin-Oakenpants** [comment]
When we get around to this ... does Onboarding still use GA?

**Comment by Thorin-Oakenpants** [comment]
now that we're on FF64, I have taken every single AS related pref and ANAL-ysed them. Results so far. Am working on the last 19

```
39 prefs: claustro's post
 3 prefs: new in 64 (asrouter.providers)
 1 pref : from elsewhere: browser.aboutHomeSnippets.updateUrl
 1 pref : from old section: browser.library.activity-stream.enabled
 9 prefs: from about:config no-one listed
==
53 prefs: total
```

<details><summary>ignore items already in user.js</summary><p>

```js
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
user_pref("browser.aboutHomeSnippets.updateUrl", "");
user_pref("browser.newtabpage.activity-stream.asrouter.providers.snippets", "");
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false);
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false);
user_pref("browser.newtabpage.activity-stream.showSponsored", false);
  //user_pref("browser.library.activity-stream.enabled", false);
```

</p></details>

<details><summary>ignore items that are local data or just UI</summary><p>

```js
user_pref("browser.newtabpage.activity-stream.showSearch", false);
user_pref("browser.newtabpage.activity-stream.feeds.topsites", false);
user_pref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeBookmarks", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeDownloads", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includeVisited", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.rows" 2);
user_pref("browser.newtabpage.activity-stream.section.topstories.rows" 1);
user_pref("browser.newtabpage.activity-stream.topSitesRows", 1);
user_pref("browser.newtabpage.activity-stream.sectionOrder" "topsites,topstories,highlights");
```

</p></details>

<details><summary>ignore cuz we like the default</summary><p>

```js
user_pref("browser.newtabpage.activity-stream.asrouter.devtoolsEnabled", false);
user_pref("browser.newtabpage.activity-stream.feeds.aboutpreferences", false);
  // ^ hides the entire Home>Firefox Home Content section
user_pref("browser.newtabpage.activity-stream.telemetry.ut.events", false);
```

</p></details>

<details><summary>ignore accounts, internals</summary><p>

```js
user_pref("browser.newtabpage.activity-stream.fxaccounts.endpoint", "");
  // ^ if you have an account & use AS then you would probably want this
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories.options", "");
  // ^ if you use pocket AND allow pocket in AS then you probably want this
user_pref("browser.newtabpage.activity-stream.feeds.migration", false);
user_pref("browser.newtabpage.activity-stream.migrationLastShownDate"
user_pref("browser.newtabpage.activity-stream.migrationRemainingDays"
user_pref("browser.newtabpage.activity-stream.migrationExpired", true);
```

</p></details>

<details><summary>ignore: not in 64 (about:config, see DXR results)</summary><p>

```js
user_pref("browser.newtabpage.activity-stream.messageCenterExperimentEnabled", false);
  // ^ zero DXR results messageCenterExperiment
user_pref("browser.newtabpage.activity-stream.showTopSites", false);
  // ^ DXR results: migrate pref code, replaced by browser.newtabpage.activity-stream.feeds.topsites
user_pref("browser.newtabpage.activity-stream.feeds.theme", false);
  //  ^ zero DXR results feeds.theme
user_pref("browser.newtabpage.activity-stream.feeds.messagecenterfeed", false);
  //  ^ zero DXR results messagecenterfeed
```

</p></details>

---

That leaves 19 items for further followup

```js
/* not in about config but.. ***/
user_pref("browser.newtabpage.activity-stream.tippyTop.service.endpoint", "");
  // ^ DXR shows a lot of "stuff*
  // ^ and way in early AS days, this was shown to be getting data and stashing it locally

/* other ***/
user_pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr"
user_pref("browser.newtabpage.activity-stream.default.sites", "");
  // ^^ 6 default sites, pretty sure the icons are already stored locally
user_pref("browser.newtabpage.activity-stream.feeds.asrouterfeed", false);
user_pref("browser.newtabpage.activity-stream.feeds.favicon", false);
user_pref("browser.newtabpage.activity-stream.feeds.newtabinit", false);
user_pref("browser.newtabpage.activity-stream.feeds.places", false);
user_pref("browser.newtabpage.activity-stream.feeds.prefs", false);
user_pref("browser.newtabpage.activity-stream.feeds.sections", false);
user_pref("browser.newtabpage.activity-stream.feeds.systemtick", false);
user_pref("browser.newtabpage.activity-stream.filterAdult", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.noDefaultSearchTile", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.havePinned"
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.searchEngines", "");
user_pref("browser.newtabpage.activity-stream.pocketCta"
user_pref("browser.newtabpage.activity-stream.prerender", false);
  // ^^ mine was already set - internal?

/* new in 64 (not exactly endpoints!) ***/
user_pref("browser.newtabpage.activity-stream.asrouter.providers.cfr", "");
  // my value contains cfr=false, it does not change with the pther cfr pref
  // `{"id":"cfr","enabled":false,"type":"local",.... blah}`
user_pref("browser.newtabpage.activity-stream.asrouter.providers.onboarding", "");
  // ^^ I thought Onboarding uses GA?
  // {"id":"onboarding","type":"local","localProvider":"OnboardingMessageProvider","enabled":true}
```

**Comment by earthlng** [comment]
tippyTop is now using "RemoteSettings" ie the same mechanism used for blocklists etc

https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/tippytop/records

These are all the buckets and collections currently used/available to RemoteSettings:

https://firefox.settings.services.mozilla.com/v1/buckets/monitor/collections/changes/records

everything with bucket `*-preview` is probably staging/development stuff and not used in production.
The interesting ones are all the "main" buckets, fe [fxmonitor-breaches](https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/fxmonitor-breaches/records?_sort=-BreachDate)

In the 1st URL above you can replace `main` with the bucket name and `tippytop` with the collection name (see 2nd link) to look at the data

**Comment by Thorin-Oakenpants** [comment]
I was also looking at all the `resource://activity-stream/` stuff.
- eg: `resource://activity-stream/data/content/tippytop/images/
 - ^^ probably used with `browser.newtabpage.activity-stream.default.sites` = YT, FB, amazon, reddit, wikipedia, twitter (well, at least for the EN portable version)

I plan on having a really good play around in a nilla 64/65. I think we're OK with everything that's left. But I do want to lock the F'er down more - also Onboarding is all local now, right, but it used to use GA. Other things to think about, the pinned, default search tiles - do they differ from the ones we sanitize ourselves - they seem built in => leads to my shit listed FollowOnSearch pet peeve.

**Comment by Thorin-Oakenpants** [comment]
OT: In my travels found this: https://github.com/mozilla/firefox-data-store-docs/blob/master/README.md . Interesting seeing the sessionstore.js/recovery.js structure .. it includes scratchpad :)

**Comment by Thorin-Oakenpants** [comment]
from the 19 currently remaining prefs
```js
user_pref("browser.newtabpage.activity-stream.improvesearch.noDefaultSearchTile", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts", false);
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.havePinned"
user_pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.searchEngines", "");
```

I think we can ignore these. I did some digging. We'll need to watch out in diffs. Seems `topSiteSearchShortcuts` is a master pref, and the whole thing is an experiment/study. Wonder if it respects Telemetry options master pref, and if that's checked if it respects experiments (I never saw anything that suggests any of that, but for all I know it's only tested on Nightly)

- `improvesearch.topSiteSearchShortcuts` (currently default false)
  * [source](https://bugzilla.mozilla.org/show_bug.cgi?id=1495517#c0): "roll out the search shortcuts experiment"
   * [source](https://hg.mozilla.org/mozilla-central/rev/66b90e5af6d162fa2c5d27b5d0df4140f108d84b#l1.12) "controls if the search shortcuts **experiment** is on"
   * ^^ so seems to be the master pref
- `improvesearch.noDefaultSearchTile` (currently default true)
   * [source(https://bugzilla.mozilla.org/show_bug.cgi?id=1484227#c0) "pref to be default true for the **62** release **study** if possible"
   * [source](https://hg.mozilla.org/integration/mozilla-inbound/rev/e6820b5308f7#l26.8) "**Experiment** to remove tiles")
- `topSiteSearchShortcuts.searchEngines` (currently google,amazon)
   * [source](https://hg.mozilla.org/integration/mozilla-inbound/rev/e6820b5308f7#l26.16) "comma-delimited list of search shortcuts that we should try and pin"
- `topSiteSearchShortcuts.havePinned`
   * I'll just guess this is set internally

Moving **three** of these to ignore, but I'll leave the master pref  while I wait for feedback/corrections

---

Also the `.filterAdult` I can't be bothered checking out, but I assume it is an option that shows when you have pocket recommended trending stories or whatever enabled (using pockets multiple categories, tags). So that can go on ignore as well. Even if it wasn't, it's clearly not privacy/security related

**Comment by Thorin-Oakenpants** [comment]
`improvesearch` - see previous pic (FF665 Beta)

pref value is "youtube, facebook, wikipedia, reddit, amazon.com, twitter.com", but note I have seven tiles, two with search overlays. That's because the `improvesearch` experiment is enabled. The improvesearch engines now only lists google (which it adds), and targets amazon (which it replaces), this is no doubt some internal cleanup - remember the 64 pref values for the search engines was `google,amazon`

To revert the improve search experiment, I have to set both `improvesearch.topSiteSearchShirtucts` **AND** `improvesearch.noDefaultSearchTile` to false, and then I'm back to 6 tiles with no search overlays

^^ so am thinking we add those two prefs to js at false as something like
```js
/* 0105e: disable Activity Stream Improve Search Experiment ***/
user_pref("improvesearch.topSiteSearchShirtucts", false);
user_pref("improvesearch.noDefaultSearchTile", false);
```

My reasoning is that these are NOT necessarily the engines YOU wanted. They're not necessarily sanitized (by yourself). Also .. its an experiment/study IMO, they use the word everywhere. That said, it's just a change that rode the trains like everything else, so they're probably using the word in a different context. IDK. I'll test the telemetry settings to see if that makes a difference (doubt it).

PS: the search bar uses your installed engine AFAIK. eg, if you change it in options to one you added yourself, it will change in newtab

**Comment by Thorin-Oakenpants** [comment]
FYI: I saw this before (last week) and then couldn't find it again. Now I have. FF65 Beta

![moar-cfr](https://user-images.githubusercontent.com/16656956/50109896-e37fcc80-0230-11e9-81fc-0f9f489a5800.png)

The GENERAL pref is `browser.newtabpage.activity-stream.asrouter.userprefs.cfr`. I had this option unchecked.

In PRIVACY&SECURITY, I checked "Allow FF to send technical data..." in order to get to the personalized extensions setting, which I then checked, and compared snapshots of prefs.js from before and after. The only difference was the master telemetry pref `datareporting.healthreport.uploadEnabled`, which is to be expected. There was also a client id pref change. The checkbox under General didn't change to match. I have no idea WTF that is all about. It doesn't seem to have a pref? Why would they even stick that there? Bugs? IDK. Something else to do for the next diffs I guess.

Really wanting to kill all endpoints now

**Comment by earthlng** [comment]
those are 2 different things. 1 is CFR, the other 1 is about personalized addon recommendations in about:addons + AMO. The latter uses the telemetry clientID and that's why it's listed there

**Comment by fmarier** [comment]
> I will assume TP works in Activity Stream (@fmarier) , and I will assume FPI works in AS

Probably not until https://bugzilla.mozilla.org/show_bug.cgi?id=1380448 is fixed.

**Comment by Thorin-Oakenpants** [comment]
:disappointed: oh well, now I better check FPI too

**Comment by Thorin-Oakenpants** [comment]
@ fmarier et al .. 

I asked Tom Ritter "does AS respect FPI" and he replied

> So Activity Stream aka about:newtab only induces network requests for background thumbnail jobs. The Pocket recommendations aren't generated by network requests.
>
> The Background Thumbnail jobs are sent cookie-less, and the Set-Cookie responses from them are thrown away. To be extra sure, the requests also come from a separate container: [1457929](https://bugzilla.mozilla.org/show_bug.cgi?id=1457929) (which will isolate all the other stuff just in case) and that container is wiped on shutdown.
> 
> So we think there's pretty good separation for Activity Stream.
>
> -tom

So "**Background Thumbnail jobs are sent cookie-less**" .. interesting. How will you apply GA then (and would it respect DNT)? In fact, how can TP work when it doesn't go through an URL Classifier / Cookie Mill? .. 

so what's this all about then:
- FF61+ [1449294](https://bugzilla.mozilla.org/show_bug.cgi?id=1449294): Capture thumbnails with tracking protection always enabled
- Open [1453448](https://bugzilla.mozilla.org/show_bug.cgi?id=1453448): Capture thumbnails with safe browsing always enabled

The (extra?) container is "nice", but I can't help but feel that there's a hole here: thumbnails from **anything** *could* end mixed up together up in that container.

Anyway, the answer to my question is "NO". FPI is not used, it uses a reserved hidden container id of 5, instead (which is inferior), and no cookies are not used (AFAICT) and since cookies aren't used there's no localStorage, IDB, but the container gets wiped. I guess thumbnails aren't wiped and don;t get an OA then.

Still, grabbing thumbnails with no control over third party connections isn't great.

**Comment by earthlng** [comment]
OT: :jeans: please add https://bugzilla.mozilla.org/1367617 to your list, thanks.

**Comment by Thorin-Oakenpants** [comment]
Where were we.. that's right .. 12 prefs to finish ANAL-ysing

`browser.newtabpage.activity-stream.prerender` - moving to ignore.  "This is set automatically by PrefsFeed.jsm" [source](https://dxr.mozilla.org/mozilla-central/source/browser/components/newtab/lib/ActivityStream.jsm#111)

---

five `activity-stream.feeds.*` prefs. See [source](https://dxr.mozilla.org/mozilla-central/source/browser/components/newtab/lib/ActivityStream.jsm) and read it, and look up any local resource:// files. While these could possibly be used to block things vs just hiding the element (e.g places->visited sites? sections->break the whole thing?), all of this data is local (bookmarks, downloads, visited sites, search tiles, default sites, search widget bar thing, etc). Fiddling with these does not improve anything privacy/security/leaky/tracking wise - and would probably just break shit. Also probably initially internal prefs initially used for testing.

Moving to ignore

```js
user_pref("browser.newtabpage.activity-stream.feeds.newtabinit", false);
  // ^ Sends a copy of the state to each new tab that is opened
user_pref("browser.newtabpage.activity-stream.feeds.places", false);
  // ^ Listens for and relays various Places-related events
user_pref("browser.newtabpage.activity-stream.feeds.prefs", false);
  // ^ dunno? preferences I guess
user_pref("browser.newtabpage.activity-stream.feeds.sections", false);
  // ^ Manages sections
user_pref("browser.newtabpage.activity-stream.feeds.systemtick", false);
  // ^ Produces system tick events to periodically check for data expiry
```

and we're down to 6

**Comment by Thorin-Oakenpants** [comment]
So we don't have to scroll up to my [constantly edited post](https://github.com/ghacksuserjs/ghacks-user.js/issues/528#issuecomment-447798611) .. I'll start a new one here for the last 6

Now we're just down to nailing external network calls - but only if they are a risk (whatever that means)
- is data 100% local (i.e packaged in each build)
- if data is not local, then when it gets it, does it use GA (I have no issues with the data itself, it would be localized bundles)
- WTF exactly is TippyTop (ELI5 please)
- when does Onboarding kick in - anytime they want to give us a new feature tour? Is it 100% fully local?

Come on guys, gals, cats ... help me finish this ticket and put everyone's minds at rest. @earthlng , you can take a break buddy (and I'll re-read your earlier comments)

6 prefs left.

```js
/* TIPPYTOP ***/
user_pref("browser.newtabpage.activity-stream.feeds.favicon", false);
  // ^ Fetches tippy top manifests from remote service
user_pref("browser.newtabpage.activity-stream.tippyTop.service.endpoint", "");
  // ^ not in about config but.. DXR shows a lot of "stuff*
  // ^ way back in early AS days, this was shown to be getting data and stashing it locally

/* ONBOARDING : I thought Onboarding uses GA ***/
user_pref("browser.newtabpage.activity-stream.asrouter.providers.cfr", ""); // FF64+
user_pref("browser.newtabpage.activity-stream.asrouter.providers.onboarding", ""); // FF64+

/* CFR ***/
user_pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr"
user_pref("browser.newtabpage.activity-stream.feeds.asrouterfeed", false);
  // ^ Handles AS Router messages, such as snippets and onboarding
```

**Comment by Thorin-Oakenpants** [comment]
Elmerghurrddd... I finally worked out WTF TippyTop is used for. We can ignore it. It's an favicon step which can reduce page thumbnails. AFAICT the process is
- checks if the website provided a favicon which has an high enough resolution (I think its 96x96?)
- verifies that there's a tippy-top fallback icon (I think it's Alexa top 200?)
- tries to request a rich-icon from FaviconFeed.jsm
- if all the above fails, it tries to get a screenshot of the page and pin it in the tile

**Comment by Thorin-Oakenpants** [comment]
Now we're talking, much more clear cut.
- IF onboarding is totally local (or if it does not use GA), then we ignore them. If we're not sure, then add them
- IF cfr is totally local, then ignore the cfr pref (and/or just rely on the pref in 5000s)

**Comment by Thorin-Oakenpants** [comment]
spent waaaay too much time on this, so I'll just take E's word for it

https://github.com/ghacksuserjs/ghacks-user.js/issues/528#issuecomment-446628055
> Onboarding and CFR are all local
